crates/cranelift/src/compiler.rs

```diff
@@ -295,23 +296,33 @@ impl wasmtime_environ::Compiler for Compiler {
         addrs.append_to(obj);
         traps.append_to(obj);
 
-        Ok(funcs.into_iter().map(|(_, f)| f.info).collect())
+        Ok((
+            funcs
+                .into_iter()
+                .zip(func_starts)
+                .map(|((_, mut f), start)| {
+                    f.info.start = start;
+                    f.info
+                })
+                .collect(),
+            trampolines,
+        ))
     }
 
     fn emit_trampoline_obj(
         &self,
         ty: &WasmFuncType,
         host_fn: usize,
         obj: &mut Object,
-    ) -> Result<()> {
+    ) -> Result<(Trampoline, Trampoline)> {
```

Comment:
Doc comment on this function indicating what the return-tuple is? I have to look at the body right now to see -- something like "Returns the trampoline in each direction: (host_to_wasm, wasm_to_host)" would help a lot.

Reply:
👍 I added this to the trait documentation in `wasmtime-environ`